### ISSUE TITLE
Chinese identifier or function causes encoding error

### ISSUE THREAD

**Issue by kevinprotoss** [Nuitka/Nuitka#832]
Hello, 

I have a problem if my python code contains Chinese words. I'll get the following error.

```
nuitka3 --module alpha/core/factor_factory.py 

Traceback (most recent call last):
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/__main__.py", line 195, in <module>
    main()
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/__main__.py", line 188, in main
    MainControl.main()
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/MainControl.py", line 749, in main
    result, options = compileTree(main_module=main_module)
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/MainControl.py", line 648, in compileTree
    makeSourceDirectory(main_module=main_module)
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/MainControl.py", line 372, in makeSourceDirectory
    writeSourceCode(filename=c_filename, source_code=source_code)
  File "/home/jwei/miniconda3/envs/python3/lib/python3.7/site-packages/nuitka/MainControl.py", line 576, in writeSourceCode
    output_file.write(source_code.encode("latin1"))
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 115417-115422: ordinal not in range(256)
```

```
nuitka3 --version
0.6.8.4
Python: 3.7.4 (default, Aug 13 2019, 20:35:49) 
Executable: /home/jwei/miniconda3/envs/python3/bin/python
OS: Linux
Arch: x86_64
```

The interesting thing is that the error does not occur for every Chinese words. If I comment some lines of codes, it will work then. However, I can not find which words are not supported, since it's just a little bit like dynamic... Sometimes the same word in other lines work, sometimes not. I have really no idea about that.

I had also tried the utf8 mode, suggested in other issues. However, it doesn't help. Does Nuitka completely support python code in other languages? Is the line for writing source file with latin-1 codec always required? I had changed the source code as the following, then it works.

```
output_file.write(source_code.encode("utf-8"))
```

Thx in advance

Best regards,
Kevin

**Comment by kevinprotoss** [comment]
I had just conclude some small code blocks. 
For example, the following two code segments. The first one does not work while the second works. 

```
def test():
    价格=0
    间隔=[2,3,4,5,6]

    for i in 间隔:
        if i==2:
            价格=1
    
    结果=价格*2
```

```
def test():
    x=0
    y=[2,3,4,5,6]

    for i in y:
        if i==2:
            x=1
    
    z=x*2
```

In addition, if I remove the `if` statement within the loop, it will also work. If I remove the last line value assignment, it will also work. That's really strange.

**Comment by kayhayen** [comment]
For C compilers, unicode is not generally the standard, and therefore Nuitka avoids producing it in the output. The generated source code, should have nothing that goes beyond "latin1". I managed to reproduce this, and it seems the unbound local error that it wants to do for one of these variables, tries to use a C string to pass the variable name, which will not work.

So that's a regression for a few releases then. I started some scalability work that should produce more efficient code, and I ought to get the variable name passed as an object, for raising the exception., with a new common helper. However, I am also disappointed to see this find any variable as potentially unassigned, the loop analysis code ought to do better than that. the `价格` ought to be know assigned and accessing it shouldn't create that code.
